packages/core/src/routes/interaction/actions/submit-interaction.ts

```diff
@@ -227,6 +229,7 @@ export default async function submitInteraction(
 
     await updateUserById(accountId, updateUserProfile);
     await assignInteractionResults(ctx, provider, { login: { accountId } });
+    ctx.assignInteractionHookDetails({ userId: accountId });
```

Comment:
you can see line 224 in the changed file?

```ts
log?.append({ userId: accountId });
```

you can assign `userId` in the same position

Reply:
I see line 224, but we cannot assign `userId` there. At that time, we don't know whether the interaction is successful.
We can only assign the `userId` when the interaction is completed.